udf-compiler/src/main/scala/com/nvidia/spark/udf/Instruction.scala

```diff
@@ -452,6 +474,91 @@ case class Instruction(opcode: Int, operand: Int, instructionStr: String) extend
     }
   }
 
+  private def arrayOp(lambdaReflection: LambdaReflection,
+      methodName: String, args: List[Expression]): Expression = {
+    // Make sure that the objref is scala.math.package$.
+    args.head match {
+      case IntegerLiteral(index) =>
+        if (!lambdaReflection.lookupField(index.asInstanceOf[Int])
+            .getType.getName.equals("scala.Array$")) {
+          throw new SparkException("Unsupported array function objref: " + args.head)
+        }
+      case _ =>
+        throw new SparkException("Unsupported array function objref: " + args.head)
+    }
+    // Translate to Catalyst
+    methodName match {
+      case "empty" =>
+        if (args.last.isInstanceOf[Repr.ClassTag[_]]) {
+          val classTag = args.last.asInstanceOf[Repr.ClassTag[_]].classTag
+          if (classTag == scala.reflect.ClassTag.Boolean) {
```

Comment:
could this `if/else` be a `match/case`?

Reply:
`} else if (classTag == scala.reflect.ClassTag("".getClass)) {` can't be converted to `case scala.reflect.ClassTag("".getClass) =>` because `scala.reflect.ClassTag("".getClass)` (https://github.com/scala/scala/blob/2.13.x/src/library/scala/reflect/ClassTag.scala#L135) returns a `GenericClassTag` object that doesn't implement `unapply` for pattern matching.

Reply:
ok fair enough, thanks for checking.